### PR TITLE
07 label_semantic_roles

### PR DESCRIPTION
**Pull request by daming-lu**
Eng & Chi README, train.py all done.

### CODE REVIEW

File: 07.label_semantic_roles/README.cn.md [hunk 1]
```diff
@@ -187,316 +187,351 @@ conll05st-release/
...
-ctx_0 = paddle.layer.data(name='ctx_0_data', type=d_type(word_dict_len))
-ctx_p1 = paddle.layer.data(name='ctx_p1_data', type=d_type(word_dict_len))
-ctx_p2 = paddle.layer.data(name='ctx_p2_data', type=d_type(word_dict_len))
-
-# 谓词上下区域标志
-mark = paddle.layer.data(name='mark_data', type=d_type(mark_dict_len))
-
-# 标注序列
-target = paddle.layer.data(name='target', type=d_type(label_dict_len))
+mark_dict_len = 2
```
Comment (jetfuel) [inline 07.label_semantic_roles/README.cn.md]: Can we keep the comments from the original versions?
```
-mark_dict_len = 2    # 谓上下文区域标志的维度，是一个0-1 2值特征，因此维度为2
-word_dim = 32        # 词向量维度
-mark_dim = 5         # 谓词上下文区域通过词表被映射为一个实向量，这个是相邻的维度
-hidden_dim = 512     # LSTM隐层向量的维度 ： 512 / 4
-depth = 8            # 栈式LSTM的深度
```

File: 07.label_semantic_roles/train.py [hunk 2]
```diff
@@ -1,228 +1,380 @@
...
 mix_hidden_lr = 1e-3
 
+IS_SPARSE = True
+PASS_NUM = 10
+BATCH_SIZE = 10
 
-def d_type(size):
-    return paddle.data_type.integer_value_sequence(size)
+embedding_name = 'emb'
+default_std = 1 / math.sqrt(hidden_dim) / 3.0  # ?
```
Comment (jetfuel) [inline 07.label_semantic_roles/train.py]: Remove # ?

File: 07.label_semantic_roles/train.py [hunk 3]
```diff
@@ -1,228 +1,380 @@
...
-    ctx_p2 = paddle.layer.data(name='ctx_p2_data', type=d_type(word_dict_len))
-    mark = paddle.layer.data(name='mark_data', type=d_type(mark_dict_len))
 
-    emb_para = paddle.attr.Param(name='emb', initial_std=0., is_static=True)
-    std_0 = paddle.attr.Param(initial_std=0.)
-    std_default = paddle.attr.Param(initial_std=default_std)
 
-    predicate_embedding = paddle.layer.embedding(
-        size=word_dim,
+# def db_lstm():
```
Comment (jetfuel) [inline 07.label_semantic_roles/train.py]: Why do we have two versions of db_lstm? Are we intend to show the v2 version as well?

File: 07.label_semantic_roles/train.py [hunk 4]
```diff
@@ -1,228 +1,380 @@
...
+        name='ctx_p2_data', shape=[1], dtype='int64', lod_level=1)
+    mark = fluid.layers.data(
+        name='mark_data', shape=[1], dtype='int64', lod_level=1)
 
     # define network topology
-    feature_out = db_lstm()
-    target = paddle.layer.data(name='target', type=d_type(label_dict_len))
-    crf_cost = paddle.layer.crf(
-        size=label_dict_len,
+    feature_out = db_lstm(**locals())
```
Comment (jetfuel) [inline 07.label_semantic_roles/train.py:122]: What is `**locals()` ? I don't see this declared elsewhere.

File: 07.label_semantic_roles/README.cn.md (hunk 1 above)
Comment (jetfuel) [inline 07.label_semantic_roles/README.cn.md]: Can we keep the original comments?
```
-mark_dict_len = 2    # 谓上下文区域标志的维度，是一个0-1 2值特征，因此维度为2
-word_dim = 32        # 词向量维度
-mark_dim = 5         # 谓词上下文区域通过词表被映射为一个实向量，这个是相邻的维度
-hidden_dim = 512     # LSTM隐层向量的维度 ： 512 / 4
-depth = 8            # 栈式LSTM的深度
```

File: 07.label_semantic_roles/README.cn.md [hunk 5]
```diff
@@ -187,316 +187,351 @@ conll05st-release/
...
-    bias_attr=std_0,
-    param_attr=lstm_para_attr)
-
-#stack L-LSTM and R-LSTM with direct edges
-input_tmp = [hidden_0, lstm_0]
-
-for i in range(1, depth):
-    mix_hidden = paddle.layer.mixed(
+def db_lstm(word, predicate, ctx_n2, ctx_n1, ctx_0, ctx_p1, ctx_p2, mark,
+            **ignored):
```
Comment (jetfuel) [inline 07.label_semantic_roles/README.cn.md]: What does the `**ignored` do? It is not used anywhere.

**Review by jetfuel — APPROVED** [review]
Some suggestions apply to Chinese, English, Train.py
Otherwise, LGTM

File: 07.label_semantic_roles/README.cn.md [hunk 6]
```diff
@@ -187,316 +187,351 @@ conll05st-release/
...
+
+                if batch_id % 10 == 0:
+                    print("avg_cost:" + str(cost))
+                    if batch_id != 0:
+                        print("second per batch: " + str((time.time(
+                        ) - start_time) / batch_id))
+                    # Set the threshold low to speed up the CI test
+                    if float(cost) < 60.0:
+                        if save_dirname is not None:
+                            # TODO(liuyiqun): Change the target to crf_decode
```
Comment (nickyfantasy) [inline 07.label_semantic_roles/README.cn.md]: remove TODO in readme? train.py we can keep it

File: 07.label_semantic_roles/train.py (hunk 4 above)
Comment (daming-lu) [reply 07.label_semantic_roles/train.py:122]:
```python
def just_print(haha, xixi, **whatever):
	print 'haha is :', haha
	print 'xixi is :', xixi

def demo_it():
	haha = 'I am just laughing'
	xixi = 'I am just grinning'
	just_print(**locals())

demo_it()	
```
Output is 
```bash
$ python show_dict.py 
haha is : I am just laughing
xixi is : I am just grinning
```

Comment (daming-lu) [reply 07.label_semantic_roles/train.py:122]: So when we call it, we pass in `locals()` which is the local variables in the current scope. `ignored` can be any word.

File: 07.label_semantic_roles/README.cn.md (hunk 5 above)
Comment (daming-lu) [reply 07.label_semantic_roles/README.cn.md]: Done

File: 07.label_semantic_roles/README.cn.md (hunk 1 above)
Comment (daming-lu) [reply 07.label_semantic_roles/README.cn.md]: Done

File: 07.label_semantic_roles/train.py (hunk 3 above)
Comment (daming-lu) [reply 07.label_semantic_roles/train.py]: Removed the commented out version

File: 07.label_semantic_roles/train.py (hunk 2 above)
Comment (daming-lu) [reply 07.label_semantic_roles/train.py]: Done

File: 07.label_semantic_roles/README.cn.md (hunk 1 above)
Comment (daming-lu) [reply 07.label_semantic_roles/README.cn.md]: Done

**Review by nickyfantasy — APPROVED** [review]

**Pull request merged**